editor/components/post-permalink/index.js

```diff
@@ -8,61 +8,88 @@ import { connect } from 'react-redux';
  */
 import { Component } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import { Dashicon, ClipboardButton, Button } from '@wordpress/components';
+import { Dashicon, Button, ClipboardButton, Tooltip } from '@wordpress/components';
 
 /**
  * Internal Dependencies
  */
 import './style.scss';
-import { isEditedPostNew, getEditedPostAttribute } from '../../store/selectors';
+import PostPermalinkEditor from './editor.js';
+import { isEditedPostNew, isPermalinkEditable, getEditedPostPreviewLink, getSamplePermalink } from '../../store/selectors';
+import { getWPAdminURL } from '../../utils/url';
 
 class PostPermalink extends Component {
 	constructor() {
 		super( ...arguments );
+
 		this.state = {
-			showCopyConfirmation: false,
+			iconClass: '',
+			editingPermalink: false,
 		};
-		this.onCopy = this.onCopy.bind( this );
-		this.onFinishCopy = this.onFinishCopy.bind( this );
-	}
-
-	componentWillUnmount() {
-		clearTimeout( this.dismissCopyConfirmation );
 	}
 
-	onCopy() {
-		this.setState( {
-			showCopyConfirmation: true,
-		} );
-	}
-
-	onFinishCopy() {
-		this.setState( {
-			showCopyConfirmation: false,
-		} );
+	componentDidUpdate( prevProps, prevState ) {
+		// If we've just stopped editing the permalink, focus on the new permalink.
+		if ( prevState.editingPermalink && ! this.state.editingPermalink ) {
+			const permalinkButton = document.querySelector( '.editor-post-permalink__link' );
+			if ( permalinkButton ) {
+				permalinkButton.focus();
+			}
+		}
 	}
 
 	render() {
-		const { isNew, link } = this.props;
-		if ( isNew || ! link ) {
+		const { isNew, previewLink, isEditable, samplePermalink } = this.props;
+		const { iconClass, editingPermalink } = this.state;
+
+		if ( isNew || ! previewLink ) {
 			return null;
 		}
 
 		return (
 			<div className="editor-post-permalink">
-				<Dashicon icon="admin-links" />
+				<Tooltip text={ __( 'Copy the permalink to your clipboard' ) }>
+					<ClipboardButton
+						className="editor-post-permalink__copy"
+						text={ samplePermalink }
+						onCopy={ () => this.setState( { iconClass: 'is-copied' } ) }
```

Comment:
Maybe we don't need `iconClass` as state. Couldn't this just be a `:focus` style for the copy button?

Reply:
This was the behaviour that @karmatosed [chose](https://github.com/WordPress/gutenberg/pull/5756#issuecomment-375635693), we can revisit it later.

Reply:
`:focus` only applies the style while the button has the focus: @karmatosed chose to have it applied from when the copy is triggered, until page reload. Sadly, there isn't a pseudo class that matches this behaviour.